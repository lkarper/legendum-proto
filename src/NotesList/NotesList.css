@import url('https://fonts.googleapis.com/css2?family=Cinzel+Decorative:wght@400;700&family=Crimson+Text&display=swap');

.NotesList__container-widget-list {
    overflow-y: auto;
    height: 100%;
    background: url('/images/paper-background-2-small.png') repeat-y center center fixed;
    padding: 10px;
    font-family: 'Crimson Text', serif;
}

.NotesList__h2-widget-list {
    margin: 5px;
}