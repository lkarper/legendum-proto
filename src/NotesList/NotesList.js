import React, { useContext, useState, useEffect } from 'react';
import UserContext from '../contexts/UserContext';
import Note from '../Note/Note';
import './NotesList.css';
import NotesListFiltersForm from '../NotesListFiltersForm/NotesListFiltersForm';

const NotesList = (props) => {

    const context = useContext(UserContext);

    const [chapterFilter, setChapterFilter] = useState('all');
    const [notesToShow, setNotesToShow] = useState(context.notes);
    const [sortType, setSortType] = useState('rec');

    useEffect(() => {
        const notes = [...context.notes];

        if (sortType === 'old') {
            notes.sort((a, b) => new Date(a.date_modified) - new Date(b.date_modified));
        } else {
            notes.sort((a, b) => new Date(b.date_modified) - new Date(a.date_modified));
        }

        if (chapterFilter === 'all') {
            setNotesToShow(notes);
        } else {
            const notesFiltered = notes.filter(note => note.exercise_id === parseInt(props.chapt));
            setNotesToShow(notesFiltered);
        }
    }, [chapterFilter, sortType, context.notes, props.chapt])

    return (
        <div className={`NotesList__container-${props.suffix}`}>
            <h2 className={`NotesList__h2-${props.suffix}`}>Saved Notes:</h2>
            <NotesListFiltersForm 
                suffix={props.suffix}
                sortType={sortType}
                chapterFilter={chapterFilter}
                setSortType={setSortType}
                setChapterFilter={setChapterFilter}
            />
            {notesToShow.length ?
                <ol>
                    {notesToShow
                        .map(note => <Note key={note.id} note={note}/> )
                    }
                </ol>
                : <p>No notes saved yet.</p>
            }
        </div>
    )
}

export default NotesList;