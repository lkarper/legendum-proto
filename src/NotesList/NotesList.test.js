import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import NotesList from './NotesList';

describe('NotesList component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<NotesList />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<NotesList />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
