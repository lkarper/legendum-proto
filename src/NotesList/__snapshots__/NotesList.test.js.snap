// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotesList component renders the UI as expected 1`] = `
<NotesList
  chapt=""
  suffix=""
>
  <div
    className="NotesList__container-"
  >
    <h3
      className="NotesList__h3-"
    >
      Saved Notes:
    </h3>
    <NotesListFiltersForm
      chapterFilter="all"
      setChapterFilter={[Function]}
      setSortType={[Function]}
      sortType="rec"
      suffix=""
    >
      <form
        className="NotesListFiltersForm__form-"
      >
        <fieldset>
          <legend>
            Sort by date saved
          </legend>
          <div>
            <input
              checked={true}
              className="NotesListFiltersForm__input radio"
              id="sort-rec"
              name="sort-notes"
              onChange={[Function]}
              type="radio"
              value="rec"
            />
            <label
              htmlFor="sort-rec"
            >
              Sort notes newest - oldest
            </label>
          </div>
          <div>
            <input
              checked={false}
              className="NotesListFiltersForm__input radio"
              id="sort-old"
              name="sort-notes"
              onChange={[Function]}
              type="radio"
              value="old"
            />
            <label
              htmlFor="sort-old"
            >
              Sort notes oldest - newest
            </label>
          </div>
        </fieldset>
      </form>
    </NotesListFiltersForm>
    <p>
      No notes saved yet.
    </p>
  </div>
</NotesList>
`;
